src/en/mangaowlto/src/eu/kanade/tachiyomi/extension/en/mangaowlto/MangaOwlTo.kt

```diff
@@ -0,0 +1,227 @@
+package eu.kanade.tachiyomi.extension.en.mangaowlto
+
+import android.app.Application
+import android.content.SharedPreferences
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.extension.en.mangaowlto.MangaOwlToFactory.Genre
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import uy.kohesive.injekt.injectLazy
+import java.util.concurrent.TimeUnit
+
+class MangaOwlTo(
+    private val collectionUrl: String,
+    extraName: String = "",
+    private val genresList: List<Genre> = listOf(),
+) : ConfigurableSource, ParsedHttpSource() {
+    override val name: String = "MangaOwl.To $extraName"
+    override val lang = "en"
+    override val supportsLatest = true
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    private val defaultDomain: String = getMirrorPref()!!
+    override val baseUrl = "https://$defaultDomain"
+    private val API = "https://api.$defaultDomain/v1"
+    private val searchPath = "search"
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        val mirrorPref = ListPreference(screen.context).apply {
+            key = MIRROR_PREF_KEY
+            title = MIRROR_PREF_TITLE
+            entries = MIRROR_PREF_ENTRIES
+            entryValues = MIRROR_PREF_ENTRY_VALUES
+            setDefaultValue(MIRROR_PREF_DEFAULT_VALUE)
+            summary = "%s"
+
+            setOnPreferenceChangeListener { _, newValue ->
+                val selected = newValue as String
+                val index = findIndexOfValue(selected)
+                val entry = entryValues[index] as String
+                preferences.edit().putString(MIRROR_PREF_KEY, entry).commit()
+            }
+        }
+        screen.addPreference(mirrorPref)
+    }
+
+    private fun getMirrorPref(): String? = preferences.getString(MIRROR_PREF_KEY, MIRROR_PREF_DEFAULT_VALUE)
```

Comment:
Inline this function into the `defaultDomain` declaration, you're using it only once.